crates/rooch-key/src/key_derive.rs

```diff
@@ -31,49 +40,291 @@ pub const DERVIATION_PATH_PURPOSE_SCHNORR: u32 = 44;
 pub const DERVIATION_PATH_PURPOSE_ECDSA: u32 = 54;
 pub const DERVIATION_PATH_PURPOSE_SECP256R1: u32 = 74;
 
-pub trait CoinOperations<Addr, KeyPair> {
-    fn derive_key_pair_from_path(
+type EncryptionKeyResult = Result<(Vec<u8>, Vec<u8>, Vec<u8>), RoochError>;
+
+pub trait CoinOperations<Addr, KeyPair, PrivKey> {
```

Comment:
The `CoinOperations` name is not suitable yet.

Reply:
Renamed.